airflow/utils/types.py

```diff
@@ -45,6 +45,7 @@ class DagRunType(str, enum.Enum):
     BACKFILL_JOB = "backfill"
     SCHEDULED = "scheduled"
     MANUAL = "manual"
+    DATASE = "dataset"
```

Comment:
a)

```suggestion
    DATASET = "dataset"
```

b) also this isn't quite the right name (as in it doesn't quite capture the meaning of what is happening). How about something like one of these

```python
    DATASET_TRIGGERED = "dataset_triggered"
    TRIGGERED = "triggered"
    DATASET_SCHEDULED = "dataset_scheduled"
```

Reply:
sounds good